src/coreclr/jit/gentree.cpp

```diff
@@ -6212,6 +6230,176 @@ unsigned Compiler::gtSetEvalOrder(GenTree* tree)
 #pragma warning(pop)
 #endif
 
+//------------------------------------------------------------------------
+// gtSetEvalOrderMinOpts: A MinOpts specific version of gtSetEvalOrder. We don't
+//    need to set costs, but we're looking for opportunities to swap operands.
+//
+// Arguments:
+//    tree - The tree for which we are setting the evaluation order.
+//
+// Return Value:
+//    the Sethi 'complexity' estimate for this tree (the higher
+//    the number, the higher is the tree's resources requirement)
+//
+unsigned Compiler::gtSetEvalOrderMinOpts(GenTree* tree)
+{
+    assert(tree);
+    if (fgOrder == FGOrderLinear)
+    {
+        // We don't re-order operands in LIR anyway.
+        return 0;
+    }
+
+    if (tree->OperIsLeaf())
+    {
+        // Nothing to do for leaves, report as having Sethi 'complexity' of 0
+        return 0;
+    }
+
+    unsigned level = 1;
+    if (tree->OperIsSimple())
+    {
+        GenTree* op1 = tree->AsOp()->gtOp1;
+        GenTree* op2 = tree->gtGetOp2IfPresent();
+
+        // Only GT_LEA may have a nullptr op1 and a non-nullptr op2
+        if (tree->OperIs(GT_LEA) && (op1 == nullptr))
+        {
+            std::swap(op1, op2);
+        }
+
+        // Check for a nilary operator
+        if (op1 == nullptr)
+        {
+            assert(op2 == nullptr);
+            return level;
+        }
+
+        if (op2 == nullptr)
+        {
+            gtSetEvalOrderMinOpts(op1);
+            return level;
+        }
+
+        level             = gtSetEvalOrderMinOpts(op1);
+        unsigned levelOp2 = gtSetEvalOrderMinOpts(op2);
+
+        bool allowSwap = true;
+        // TODO: Introduce a function to check whether we can swap the order of its operands or not.
+        switch (tree->OperGet())
+        {
+            case GT_COMMA:
+            case GT_BOUNDS_CHECK:
+            case GT_INTRINSIC:
+            case GT_QMARK:
+            case GT_COLON:
+                // We're not going to swap operands in these
+                allowSwap = false;
+                break;
+
+            case GT_STORE_BLK:
+            case GT_STOREIND:
+            {
+                if (op1->IsInvariant())
+                {
+                    allowSwap = false;
+                    tree->SetReverseOp();
+                    break;
+                }
+                if ((op1->gtFlags & GTF_ALL_EFFECT) != 0)
+                {
+                    break;
+                }
+
+                // In case op2 assigns to a local var that is used in op1, we have to evaluate op1 first.
+                if (gtMayHaveStoreInterference(op2, op1))
```

Comment:
You probably don't need this expensive version in MinOpts. It can be the conservative `(op2->gtFlags & GTF_ASG) != 0` that existed before #97409.

Reply:
> You probably don't need this expensive version in MinOpts. It can be the conservative `(op2->gtFlags & GTF_ASG) != 0` that existed before #97409.

just tried - it adds +28k bytes size regression for `benchmarks.run_pgo` collection

Reply:
Ah ok, fine to leave it then. (Actually looks like I called out explicitly doing it in MinOpts in that PR as well)